pkg/autoscaler/autoscaler/pdplan_autoscaler.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaler
+
+import (
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	"github.com/pingcap/tidb-operator/pkg/pdapi"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const groupLabelKey = "group"
+
+func (am *autoScalerManager) syncPlans(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, plans []pdapi.Plan) error {
+	planGroups := sets.String{}
+	groupPlanMap := make(map[string]pdapi.Plan)
+	for _, plan := range plans {
+		groupName := plan.Labels[groupLabelKey]
+		planGroups.Insert(groupName)
+		groupPlanMap[groupName] = plan
+	}
+
+	// Filter all autoscaled TidbClusters
+	requirement, err := labels.NewRequirement(label.AutoScalingGroupLabelKey, selection.Exists, nil)
+	if err != nil {
+		return err
+	}
+	selector := labels.NewSelector().Add(*requirement)
+	tcList, err := am.tcLister.List(selector)
+	if err != nil {
+		return err
+	}
+
+	existedGroups := sets.String{}
+	groupTcMap := make(map[string]*v1alpha1.TidbCluster)
+	for _, tc := range tcList {
+		groupName := tc.Labels[label.AutoScalingGroupLabelKey]
```

Comment:
Consider the case that the label does not exist?

Reply:
The label selector makes sure it exist